scipy/spatial/qhull.pyx

```diff
@@ -2405,10 +2516,244 @@ class Voronoi(_QhullUser):
         self._ridge_dict = None
 
         _QhullUser._update(self, qhull)
+        self.points = self._points
+
+    def add_points(self, points, restart=False):
+        _QhullUser._add_points(self, points, restart)
 
     @property
     def ridge_dict(self):
         if self._ridge_dict is None:
             self._ridge_dict = dict(zip(map(tuple, self.ridge_points.tolist()),
                                         self.ridge_vertices))
         return self._ridge_dict
+
+Voronoi.add_points.__func__.__doc__ = _QhullUser._add_points.__doc__
+
+#------------------------------------------------------------------------------
+# Halfspace Intersection
+#------------------------------------------------------------------------------
+
+class HalfspaceIntersection(_QhullUser):
+    """
+    HalfspaceIntersection(halfspaces, feasible_point, incremental=False, qhull_options=None)
+
+    Halfspace intersections in N dimensions.
+
+    .. versionadded:: 0.19.0
+
+    Parameters
+    ----------
+    halfspaces : ndarray of floats, shape (nineq, ndim+1)
+        Stacked Inequalities of the form Ax + b <= 0 in format [A; b]
+    feasible_point: ndarray of floats, shape (ndim,)
+        Feasible point inside the region defined by halfspaces.
+        It can be obtained by linear programming.
+    incremental : bool, optional
+        Allow adding new halfspaces incrementally. This takes up some additional
+        resources.
+    qhull_options : str, optional
+        Additional options to pass to Qhull. See Qhull manual
+        for details. (Default: "Qx" for ndim > 4 and "" otherwise)
+        Option "H" is always enabled.
+
+    Attributes
+    ----------
+    halfspaces : ndarray of double, shape (nineq, ndim+1)
+        Input halfspaces.
+    intersections : ndarray of double, shape (ninter, ndim)
+        Intersections of all halfspaces.
+    dual_points : ndarray of double, shape (nineq, ndim)
+        Dual points of the input halfspaces.
+    dual_simplices : list of lists of ints
+        Indices of points forming the simplical facets of the dual convex hull.
```

Comment:
I think the facets are not necessarily simplical --- maybe rename to `dual_facets` etc.